evalml/tests/automl_tests/test_automl_iterative_algorithm.py

```diff
@@ -0,0 +1,1146 @@
+        max_batches = 1
+        # _automl_algorithm will include all allowed_pipelines in the first batch even
+        # if they are not searched over. That is why n_automl_pipelines does not equal
+        # n_results when max_iterations and max_batches are None
+        n_automl_pipelines = 1 + len(automl.allowed_pipelines)
+        num_ensemble_batches = 0
+    else:
+        # automl algorithm does not know about the additional stacked ensemble pipelines
+        num_ensemble_batches = (
+            (max_batches - 1) // ensemble_nth_batch if use_ensembling else 0
+        )
+        # So that the test does not break when new estimator classes are added
+        n_results = (
+            1
+            + len(automl.allowed_pipelines)
+            + (5 * (max_batches - 1 - num_ensemble_batches))
+            + num_ensemble_batches
+        )
+        n_automl_pipelines = n_results
+    assert automl._automl_algorithm.batch_number == max_batches
+    assert automl._automl_algorithm.pipeline_number + 1 == n_automl_pipelines
+    assert len(automl.results["pipeline_results"]) == n_results
+    if num_ensemble_batches == 0:
+        assert automl.rankings.shape[0] == min(
+            1 + len(automl.allowed_pipelines), n_results
+        )  # add one for baseline
+    else:
+        assert automl.rankings.shape[0] == min(
+            2 + len(automl.allowed_pipelines), n_results
+        )  # add two for baseline and stacked ensemble
+    assert automl.full_rankings.shape[0] == n_results
+
+
+@patch("evalml.tuners.skopt_tuner.SKOptTuner.add")
+def test_pipeline_hyperparameters_make_pipeline_other_errors_iterative(
+    mock_add, AutoMLTestEnv, X_y_multi
+):
+    X, y = X_y_multi
+    custom_hyperparameters = {
+        "Imputer": {"numeric_impute_strategy": Categorical(["most_frequent", "mean"])}
+    }
+    estimators = get_estimators("multiclass", [ModelFamily.EXTRA_TREES])
+
+    component_graphs = {}
+    for ind, estimator in enumerate(estimators):
+        component_graphs[f"CG_{ind}"] = [estimator]
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="multiclass",
+        allowed_component_graphs=component_graphs,
+        custom_hyperparameters=custom_hyperparameters,
+        n_jobs=1,
+        _automl_algorithm="iterative",
+    )
+    env = AutoMLTestEnv("multiclass")
+
+    mock_add.side_effect = ValueError("Alternate error that can be thrown")
+    with pytest.raises(ValueError) as error:
+        with env.test_context(score_return_value={"Log Loss Multiclass": 1.0}):
+            automl.search()
+    assert "Alternate error that can be thrown" in str(error.value)
+    assert "Default parameters for components" not in str(error.value)
+
+
+@pytest.mark.parametrize("component_graphs", [True, False])
+@pytest.mark.parametrize("automl_parameters", [True, False])
+@pytest.mark.parametrize("custom_hyperparameters", [True, False])
+def test_pipeline_custom_hyperparameters_make_pipeline_iterative(
```

Comment:
These couple of tests use `allowed_component_graphs` to test custom hyperparameter and pipeline parameter logic. We could change these tests to be algorithm agnostic but I believe:
1. These tests should be tested on the algorithm side
2. We already have coverage for both algorithms on the algorithm side.